src/librados/librados.cc

```diff
@@ -4531,17 +4554,26 @@ extern "C" int rados_watch2(rados_ioctx_t io, const char *o, uint64_t *handle,
     librados::IoCtxImpl *ctx = (librados::IoCtxImpl *)io;
     object_t oid(o);
     C_WatchCB2 *wc = new C_WatchCB2(watchcb, watcherrcb, arg);
-    ret = ctx->watch(oid, cookie, NULL, wc, true);
+    ret = ctx->watch(oid, cookie, NULL, wc, timeout, true);
   }
-  tracepoint(librados, rados_watch_exit, ret, handle ? *handle : 0);
+  tracepoint(librados, rados_watch3_exit, ret, handle ? *handle : 0);
   return ret;
 }
 
 extern "C" int rados_aio_watch(rados_ioctx_t io, const char *o,
                                rados_completion_t completion,
                                uint64_t *handle,
                                rados_watchcb2_t watchcb,
-                               rados_watcherrcb_t watcherrcb, void *arg)
+                               rados_watcherrcb_t watcherrcb, void *arg) {
+  return rados_aio_watch2(io, o, completion, handle, watchcb, watcherrcb, 0, arg);
+}
+
+extern "C" int rados_aio_watch2(rados_ioctx_t io, const char *o,
+                                rados_completion_t completion,
+                                uint64_t *handle,
+                                rados_watchcb2_t watchcb,
+                                rados_watcherrcb_t watcherrcb,
+                                uint32_t timeout, void *arg)
 {
   tracepoint(librados, rados_aio_watch_enter, io, o, completion, handle, watchcb, arg);
```

Comment:
now that `rados_aio_watch2()` is delegated to do the work. we can probably remove `rados_aio_watch_enter` and `rados_aio_watch_exit` tracing points, and add `rados_aio_watch2_enter` and `rados_aio_watch2_exit` to `rados_aio_watch2()`.